rita_common/src/network_monitor/mod.rs

```diff
@@ -102,84 +83,62 @@ pub struct IfaceStats {
     packet_loss: PacketLossStats,
 }
 
-impl Message for GetStats {
-    type Result = Result<Stats, RitaCommonError>;
-}
-
 pub type Stats = HashMap<String, IfaceStats>;
 
-impl Handler<GetStats> for NetworkMonitor {
-    type Result = Result<Stats, RitaCommonError>;
-
-    fn handle(&mut self, _msg: GetStats, _ctx: &mut Context<Self>) -> Self::Result {
-        let mut stats = Stats::new();
-
-        for (iface, latency_stats) in self.latency_history.iter() {
-            if let Some(packet_loss_stats) = self.packet_loss_history.get(iface) {
-                stats.insert(
-                    iface.clone(),
-                    IfaceStats {
-                        latency: LatencyStats {
-                            avg: latency_stats.get_avg(),
-                            std_dev: latency_stats.get_std_dev(),
-                        },
-                        packet_loss: PacketLossStats {
-                            avg: packet_loss_stats.get_avg(),
-                            five_min_avg: packet_loss_stats.get_five_min_average(),
-                        },
+pub fn get_stats(_msg: GetStats) -> Stats {
+    let mut stats = Stats::new();
+
+    for (iface, latency_stats) in (*(NETWORK_MONITOR.write().unwrap())).latency_history.iter() {
+        if let Some(packet_loss_stats) = (*(NETWORK_MONITOR.write().unwrap())).packet_loss_history.get(iface) {
```

Comment:
here's probably your deadlock, get the data from network monitor once at the top, then use that value twice here. Each time you call 'write' you're grabbing a write lock, a write lock is exclusive so right now you are grabbing one once at the start of the for loop and again during each iteration, resulting in a deadlock. 

These can be a lot more subdle, hopefully this is it and we have an easy time of it.